### ISSUE TITLE
2.9.5 with docker and python 3

### ISSUE THREAD

**Issue by priest110** [ckan/ckan#6873]
2.9.5

How can I use ckan installed with docker and python 3, since with the latest stable version (2.9.5) it has docker environment using python 2?

**Comment by priest110** [comment]
Now I am using the master branch, but I still have errors: initially, datapusher doesn't work, giving 'Process completed but unable to post to result_url', solved with this [solution](https://github.com/ckan/ckan/issues/3987#issuecomment-359486018). But now I have a new error, 403 error (JobError: CKAN DataStore bad response. Status code: 403 FORBIDDEN. At: http://dockerhost:5000/api/3/action/datastore_create. Response: {u'error': {u'__type': u'Authorization Error', u'message': u'Access denied: Action datastore_create requires an authenticated user'},  u'help': u'http://dockerhost:5000/api/3/action/help_show?name=datastore_create', u'success': False}). What I can do?

**Comment by kowh-ai** [comment]
Can you please post the curl command you are using?

**Comment by priest110** [comment]
For example: curl -X POST http://dockerhost:5000/api/3/action/datastore_create -H "Authorization: MY-API-KEY" -d '{"resource": {"package_id": "SOME-PACKAGE-ID"}, "fields": {"id": "a"}, {"id": "b"} ], "records": [ {"a": 1, "b": "xyz"}, {"a": 2, "b": "zzz"} ]} ]}'. I found that if I replaced [this]( https://github.com/ckan/ckan/blob/91c9366cf8df0dc4e12a9a043756eda5e40534b0/ckan/lib/api_token.py#L83) with MY-API-KEY and it works, so I believe the problem is related to that.

**Comment by kowh-ai** [comment]
Not sure what the problem is. If you replace **MY-API-KEY** with your actual API key you have generated with your user in the GUI _and_ **SOME-PACKAGE-ID** with your actual Package ID you have already created, then this works. I just tested this with a fresh Docker build of the master repo

**Comment by priest110** [comment]
Right, when I say MY-API-KEY is actually my API key that I generated with my user in the GUI and SOME-PACKAGE-ID is actually a package id that I already created. And does this Docker build have the datapusher working well?

**Comment by priest110** [comment]
Ok, I found that the user that is not authenticated is the default user (named after ckan.site_id in the configuration file) that runs internal tasks. So how can I fix this?
